src/jenkins_job_insight/storage.py

```
@@ -549,6 +1521,297 @@ async def save_html_report(job_id: str, html_content: str) -> Path:
+        cursor = await db.execute(
+            f"SELECT COUNT(*) FROM failure_history WHERE {where}",
+            params,
+        )
+        total = (await cursor.fetchone())[0]
+
+        # Get paginated results
+        db.row_factory = aiosqlite.Row
+        cursor = await db.execute(
+            f"SELECT id, job_id, job_name, build_number, test_name, error_message, "
+            f"error_signature, classification, child_job_name, child_build_number, analyzed_at "
+            f"FROM failure_history WHERE {where} ORDER BY analyzed_at DESC LIMIT ? OFFSET ?",
```

Ruff (0.15.6)

[error] Possible SQL injection vector through string-based query construction

(S608)